pkg/kubelet/cm/cpumanager/cpu_manager.go

```diff
@@ -328,13 +329,17 @@ func (m *manager) updateContainerCPUSet(p *v1.Pod, containerID string, cpus cpus
 			})
 	}
 
-	runtimeService, err := m.runtimeManager.GetRuntimeServiceByPod(p)
+	podWorkloadType := runtimeregistry.ContainerWorkloadType
+	if p.Spec.VirtualMachine != nil {
+		podWorkloadType = runtimeregistry.VmworkloadType
+	}
+	runtimeService, err := m.runtimeManager.GetRuntimeServiceByWorkloadType(podWorkloadType)
```

Comment:
Why not have a method like GetRuntimeServiceByPod()? It should be pretty common to check if VM or container and then return a runtime.